### DISCUSSION TITLE
Cannot seem to enable acks_late with Celery and SQS

### DISCUSSION THREAD

**Discussion by ZPerrin** [celery/celery#8418] [Q&A] [answered]
Hey everyone,

I've been struggling with this issue for the last few days and am at my wits end.

I am using the latest version of celery in combination with AWS ECS and SQS:
`celery[sqs]<=5.3.1`

**** EDIT **** 
This actually presents in `celery[sqs]<=5.2.7` as a dependency
**** EDIT ****

I cannot for the life of me force my workers to use a late acknowledgement to SQS, though they are able to connect to SQS just fine and process tasks accordingly.

Here is what I have tried:

### python config ###
```
celery_sqs_app = Celery(
    'audio_transcription_celery_sqs_app',
    broker="sqs://",
    include=['src.celery.tasks']
)

celery_sqs_app.conf.update(
    task_default_queue='audio_transcription_queue.fifo',
    task_acks_late=True,
    task_reject_on_worker_lost=True,
    worker_prefetch_multiplier=1,
    broker_transport_options={
        'region': constants.AWS_DEFAULT_REGION,
        'polling_interval': 30,
        'predefined_queues': {
            'audio_transcription_queue.fifo': {
                'url': sqs_url
            }
        }
    }
)
```

This is referenced when starting my worker via supervisord like so: 
`celery -A src.celery.celery_config:celery_sqs_app worker --prefetch-multiplier=1 --loglevel=info --concurrency=1 -E`

My single task is defined in src.celery.tasks and it's signature looks like this:
```
@celery_sqs_app.task(bind=True, base=TranscriptionBaseTask, soft_time_limit=1800, time_limit=1920, acks_late=True)
def transcribe_audio_task(self, request_payload):
```

My issue is that i'm using a auto-scaling scheme and these tasks are long running, on a scale down operation I would like the tasks terminated via SIGKILL (fargate limits us to 120 graceperiod for shutdown) to be re-queued via the SQS visibility timeout which i have properly configured.  However my observations show that no matter what, my workers are acking immediately.

I can see that the pre-fetch multiplier is working correctly as at least the pre-fetched messages remain as 'in-flight' in the SQS ui, but the worked messages still seem to be immediately acknowledged and removed from the queue.

Any help to point me in the right direction here would be appreciated.  Is this a limitation of SQS?

**Comment by auvipy** [comment]
can you confirm celery[sqs]<=5.3.1 this command is working? it didn't work in my terminal

**Reply by ZPerrin** [reply]
Thanks for reaching out.

My apologies, the failure to honor the late acknowledgement settings was presenting for me with
`celery[sqs]<=5.2.7`

You're right, the 5.3.1 syntax doesn't work with pip. I'm trying to update to 5.3.1 to see if that fixes the issue, but specifying the version in the bundle isn't playing well with pip for some reason. The above syntax for 5.2.7 works for me.

I'm now attempting to skip the bundle and just install the dependencies manually.

**Reply by ZPerrin** [reply, marked as answer]
After some further digging I believe I've misdiagnosed the issue being related to not honoring late acknowledgments.

I appear to be running into the same issue as others where workers are randomly unable to connect to SQS: https://github.com/celery/celery/issues/8243

The reason this was obscure at first was because i'm using an auto-scaling ECS target group that adds new containers running celery into the pool periodically based on SQS queue length.

What i'm noticing, is that as new servers come online and connect, they are able to start receiving requests.  The older servers in turn start getting empty message responses from SQS, which has the side effect of my in-flight messages from SQS always remaining the same (the expected values given a single server worker pool).

I'm going to move my discussion to the above ticket.
